### PR TITLE
Enhance A11y for Playlist

### PR DESCRIPTION
**Pull request by sangwoo108**
* Add a11y texts for labeless buttons
* add tab-index for components
* Replace contextual menu with `buttonMenu` from Nala - which allows us to have TABable items in the menu.
* Replace some buttons with Nala button.

Resolves https://github.com/brave/brave-browser/issues/32600

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by mkarolin — APPROVED** [review]
`strings`++

File: components/playlist/browser/resources/components/contextualMenu.tsx [hunk 1]
```diff
@@ -126,27 +46,27 @@ export default function ContextualMenuAnchorButton ({
...
 }: MenuProps) {
-  const [showingMenu, setShowingMenu] = React.useState(false)
+  const menuButtonProps: ButtonMenuProps = {}
+  // Force menu widget to be closed when the anchor button is not visible. In
+  // case where it's visible, the menuButtonProps doesn't contain isOpen property,
+  // so it won't affect the behavior of the menu.
+  if (!visible) menuButtonProps.isOpen = false
+
+  // TODO(sko) We don't have event for opening menu widget. Once it's ready,
+  // wire onShowMenu and onDismissMenu to corresponding events.
```
Comment (fallaciousreasoning) [inline components/playlist/browser/resources/components/contextualMenu.tsx:56]: cc @nullhook we should add these!

File: components/playlist/browser/resources/components/contextualMenu.tsx [hunk 2]
```diff
@@ -126,27 +46,27 @@ export default function ContextualMenuAnchorButton ({
   onShowMenu,
   onDismissMenu
 }: MenuProps) {
-  const [showingMenu, setShowingMenu] = React.useState(false)
+  const menuButtonProps: ButtonMenuProps = {}
+  // Force menu widget to be closed when the anchor button is not visible. In
+  // case where it's visible, the menuButtonProps doesn't contain isOpen property,
+  // so it won't affect the behavior of the menu.
+  if (!visible) menuButtonProps.isOpen = false
```
Comment (fallaciousreasoning) [inline components/playlist/browser/resources/components/contextualMenu.tsx:53]: Does this actually do anything?

File: components/playlist/browser/resources/components/contextualMenu.tsx [hunk 3]
```diff
@@ -126,27 +46,27 @@ export default function ContextualMenuAnchorButton ({
...
-          onClickOutside={() => {
-            setShowingMenu(false)
-            if (onDismissMenu) onDismissMenu()
-          }}
-        />
-      )}
-    </StyledAnchorButton>
+      </div>
+      {items.map(i => (
+        <leo-menu-item key={i.name} onClick={i.onClick}>
```
Comment (fallaciousreasoning) [inline components/playlist/browser/resources/components/contextualMenu.tsx:63]: Note: You **might** be able to do 

```tsx
const StyledRow = styled('leo-menu-item')`
    foo: bar;
`
```

and not have a separate StyledRow and `leo-menu-item` here

File: components/playlist/browser/resources/components/playlistItem.tsx [hunk 4]
```diff
@@ -351,8 +351,18 @@ export const SortablePlaylistItem = React.forwardRef(
...
-      <div ref={setNodeRef} style={style} {...attributes} {...listeners}>
-        <a ref={forwardedRef} href={`#${itemId}`} />
+      <div
+        ref={setNodeRef}
+        style={style}
+        {...attributes}
+        {...listeners}
+        tabIndex={-1} // disables tab traversal for sortable as we have our own tab traversal logic
+      >
+        <a
```
Comment (fallaciousreasoning) [inline components/playlist/browser/resources/components/playlistItem.tsx:361]: Maybe this should be aria-hidden?

**Review by fallaciousreasoning — APPROVED** [review]
LGTM

File: components/playlist/browser/resources/components/contextualMenu.tsx (hunk 2 above)
Comment (sangwoo108) [reply components/playlist/browser/resources/components/contextualMenu.tsx:53]: Yeah, when mouse leaves items, the menu will be hidden. When mouse came back into the item, menu is still open without this.

File: components/playlist/browser/resources/components/playlistItem.tsx (hunk 4 above)
Comment (sangwoo108) [reply components/playlist/browser/resources/components/playlistItem.tsx:362]: Good call!

File: components/playlist/browser/resources/components/contextualMenu.tsx (hunk 3 above)
Comment (sangwoo108) [reply components/playlist/browser/resources/components/contextualMenu.tsx]: Nice! I didn't even try that 👍

**Pull request merged**